CIScripts/Test/Utils/ComponentsInstallation.ps1

```diff
@@ -0,0 +1,64 @@
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+
+function Invoke-MsiExec {
+    Param (
+        [Switch] $Uninstall,
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session,
+        [Parameter(Mandatory = $true)] [String] $Path
+    )
+
+    $Action = if ($Uninstall) { "/x" } else { "/i" }
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        $Result = Start-Process msiexec.exe -ArgumentList @($Using:Action, $Using:Path, "/quiet") `
+            -Wait -PassThru
+        if ($Result.ExitCode -ne 0) {
+            throw "Installation of $Using:Path failed with $($Result.ExitCode)"
+        }
+
+        # Refresh Path
+        $Env:Path = [System.Environment]::GetEnvironmentVariable("Path", "Machine")
+    }
+}
+
+function Install-Agent {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Installing Agent"
+    Invoke-MsiExec -Session $Session -Path "C:\Artifacts\contrail-vrouter-agent.msi"
+}
+
+function Uninstall-Agent {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Uninstalling Agent"
+    Invoke-MsiExec -Uninstall -Session $Session -Path "C:\Artifacts\contrail-vrouter-agent.msi"
+}
+
+function Install-Extension {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Installing vRouter Forwarding Extension"
+    Invoke-MsiExec -Session $Session -Path "C:\Artifacts\vRouter.msi"
+}
+
+function Uninstall-Extension {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Uninstalling vRouter Forwarding Extension"
+    Invoke-MsiExec -Uninstall -Session $Session -Path "C:\Artifacts\vRouter.msi"
+}
+
+function Install-Utils {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Installing vRouter utility tools"
+    Invoke-MsiExec -Session $Session -Path "C:\Artifacts\utils.msi"
+}
+
+function Uninstall-Utils {
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    Write-Host "Uninstalling vRouter utility tools"
+    Invoke-MsiExec -Uninstall -Session $Session -Path "C:\Artifacts\utils.msi"
+}
```

Comment:
Did you mean to copy these functions from `CIScripts/Test/Utils/CommonTestCode.ps1` or move them?

Reply:
Also, why move them out from `CommonTestCode`? Is your plan to split this file?

Reply:
Oversight on my part, I wanted to move those functions to separate file.

Reply:
> Also, why move them out from CommonTestCode? Is your plan to split this file?

yes. They form a thematic subgroup which is not related to any other functions in this file. Common*, Utils*, Misc* is very often an antipattern.

Reply:
E.g. if I want to install stuff using MSI, I have to import the whole file that has functions that are useless to me, like pinging containers. CRP is violated:
```
Common-Reuse Principle (CRP)
The CRP states that classes that tend to be reused together belong in the same package together. It is a way of helping to decide which classes belong in which package.
When depending on a package, one want to make sure that the classes are inseparable and interdependent, which is also handy when culling classes that do not belong into the package.
```
https://en.wikipedia.org/wiki/Package_principles